packages/contracts-bedrock/src/L1/OPStackManager.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ISemver } from "src/universal/ISemver.sol";
+import { SystemConfig } from "src/L1/SystemConfig.sol";
+
+contract OPStackManager is ISemver {
+    /// @custom:semver 1.0.0-beta.1
+    string public constant version = "1.0.0-beta.1";
+
+    /// @notice Represents the roles that can be set when deploying a standard OP Stack chain.
+    struct Roles {
+        address proxyAdminOwner;
+        address systemConfigOwner;
+        address batcher;
+        address unsafeBlockSigner;
+        address proposer;
+        address challenger;
+    }
+
+    /// @notice Emitted when a new OP Stack chain is deployed.
+    /// @param l2ChainId The chain ID of the new chain.
+    /// @param systemConfig The address of the new chain's SystemConfig contract.
+    event Deployed(uint256 indexed l2ChainId, SystemConfig indexed systemConfig);
+
+    /// @notice Thrown when an invalid `l2ChainId` is provided to `deploy`.
+    error InvalidChainId();
+
+    /// @notice Thrown when a deployment fails.
+    error DeploymentFailed(string reason);
+
+    function deploy(
+        uint256 l2ChainId,
+        Roles calldata roles,
```

Comment:
Nit: I generally have a preference for value types to precede reference types in function declarations.

This could be addressed by defining a `FeeScalar` struct.

Reply:
What do you think about instead switching the order to move `roles` last? Structs can be a bit cumbersome sometimes and since there's only two fee scalars I think we can avoid them. If that works, I will get a PR in to the spec repo to update the function signature https://specs.optimism.io/experimental/op-stack-manager.html#deploy

Reply:
I'm totally fine with that approach.

Reply:
Resolving per https://github.com/ethereum-optimism/specs/pull/302